### PR TITLE
feat: enable feature for creating a development image

### PR DESCRIPTION
**Pull request by joshuachp**
Enable services and options to create a development image:

The packages installed are:

- `dropbear`: ssh client for embedded
- `avahi`: for the mDNS to connect to the Pi using the hostname
- `docker`: as the container runtime

We needed to modify some option to make the services work:

- Enable root login and give a default password to connect through ssh
- Link the `/otp` and `/etc/docker` paths to writable partitions, since the docker daemon needed to write the `.tmpkey.json` into that path and `containerd` writes it data into `/opt/containerd`. It can fail if the directory `containerd` already exists.
- Increased the `data` partition to **2 GB**, since we are working with a Raspberry Pi in a dev environment I think it's an acceptable size to store containers and data. 
  Usually an SD card is at least 4GBs and this still leaves us with more or less 1 GBs of OTA space. Although an 8GB SD card would be better.

The rootfs should be increased manually through `make menuconfig` to an appropriate size, we didn't pick a default one since the package sizes could change in the future, but `512M` is more than sufficient.

### CODE REVIEW

**Comment by joshuachp** [comment]
Some issues with this setup is that the `dockerd.json` file is in the data partition, so I believe it cannot be updated through Rauc.

Another note is that bmap-tools has issues creating the image on an encrypted fs.

**Review by rgallor — APPROVED** [review]

File: configs/raspberrypi4-64-rauc_defconfig [hunk 1]
```diff
@@ -58,3 +58,13 @@ BR2_PACKAGE_HOST_UBOOT_TOOLS_ENVIMAGE_SIZE="0x8000"
 BR2_PACKAGE_HOST_UBOOT_TOOLS_ENVIMAGE_REDUNDANT=y
 BR2_PACKAGE_HOST_UBOOT_TOOLS_BOOT_SCRIPT=y
 BR2_PACKAGE_HOST_UBOOT_TOOLS_BOOT_SCRIPT_SOURCE="$(BR2_EXTERNAL_BR2RAUC_PATH)/board/raspberrypi4-64/u-boot_4_64bit.ush"
+# Docker
+BR2_PACKAGE_DROPBEAR=y
+BR2_PACKAGE_DOCKER_CLI=y
+BR2_PACKAGE_DOCKER_CLI_STATIC=y
+BR2_PACKAGE_DOCKER_COMPOSE=y
```
Comment (harlem88) [inline configs/raspberrypi4-64-rauc_defconfig:65]: Do we still need docker compose? Maybe not, this is something that will no longer be supported.

File: configs/raspberrypi_defconfig [hunk 2]
```diff
@@ -61,3 +61,12 @@ BR2_PACKAGE_HOST_UBOOT_TOOLS_ENVIMAGE_SIZE="0x8000"
 BR2_PACKAGE_HOST_UBOOT_TOOLS_ENVIMAGE_REDUNDANT=y
 BR2_PACKAGE_HOST_UBOOT_TOOLS_BOOT_SCRIPT=y
 BR2_PACKAGE_HOST_UBOOT_TOOLS_BOOT_SCRIPT_SOURCE="$(BR2_EXTERNAL_BR2RAUC_PATH)/board/raspberrypi/u-boot.ush"
+# Docker
+BR2_PACKAGE_DROPBEAR=y
+BR2_PACKAGE_DOCKER_CLI=y
+BR2_PACKAGE_DOCKER_CLI_STATIC=y
+BR2_PACKAGE_DOCKER_COMPOSE=y
```
Comment (harlem88) [inline configs/raspberrypi_defconfig:68]: same above

File: configs/raspberrypicm4io-64-rauc_defconfig [hunk 3]
```diff
@@ -58,3 +58,12 @@ BR2_PACKAGE_HOST_UBOOT_TOOLS_ENVIMAGE_SIZE="0x8000"
 BR2_PACKAGE_HOST_UBOOT_TOOLS_ENVIMAGE_REDUNDANT=y
 BR2_PACKAGE_HOST_UBOOT_TOOLS_BOOT_SCRIPT=y
 BR2_PACKAGE_HOST_UBOOT_TOOLS_BOOT_SCRIPT_SOURCE="$(BR2_EXTERNAL_BR2RAUC_PATH)/board/raspberrypicm4io-64/u-boot_cm4io_64bit.ush"
+# Docker
+BR2_PACKAGE_DROPBEAR=y
+BR2_PACKAGE_DOCKER_CLI=y
+BR2_PACKAGE_DOCKER_CLI_STATIC=y
+BR2_PACKAGE_DOCKER_COMPOSE=y
```
Comment (harlem88) [inline configs/raspberrypicm4io-64-rauc_defconfig:65]: same above

**Review by harlem88 — CHANGES_REQUESTED** [review]

File: configs/raspberrypi4-64-rauc_defconfig (hunk 1 above)
Comment (joshuachp) [reply configs/raspberrypi4-64-rauc_defconfig]: Done

File: configs/raspberrypi_defconfig (hunk 2 above)
Comment (joshuachp) [reply configs/raspberrypi_defconfig]: Done

File: configs/raspberrypicm4io-64-rauc_defconfig (hunk 3 above)
Comment (joshuachp) [reply configs/raspberrypicm4io-64-rauc_defconfig]: Done

**Review by harlem88 — APPROVED** [review]

**Pull request merged**